commit ecdc0b1def603378d21488ca7f7def09f3544c99
Author: Aitor <aitor.rufian@alu.uclm.es>
Date:   Thu Apr 16 22:28:42 2026 +0200

    fix(pipewire): avoid crash on device disconnect in default tracker

diff --git a/src/services/pipewire/defaults.hpp b/src/services/pipewire/defaults.hpp
index f31669e..8881ca0 100644
--- a/src/services/pipewire/defaults.hpp
+++ b/src/services/pipewire/defaults.hpp
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <qobject.h>
+#include <qpointer.h>
 #include <qtmetamacros.h>
 
 #include "registry.hpp"
@@ -67,16 +68,16 @@ private:
 	PwRegistry* registry;
 	PwBindableRef<PwMetadata> defaultsMetadata;
 
-	PwNode* mDefaultSink = nullptr;
+	QPointer<PwNode> mDefaultSink = nullptr;
 	QString mDefaultSinkName;
 
-	PwNode* mDefaultSource = nullptr;
+	QPointer<PwNode> mDefaultSource = nullptr;
 	QString mDefaultSourceName;
 
-	PwNode* mDefaultConfiguredSink = nullptr;
+	QPointer<PwNode> mDefaultConfiguredSink = nullptr;
 	QString mDefaultConfiguredSinkName;
 
-	PwNode* mDefaultConfiguredSource = nullptr;
+	QPointer<PwNode> mDefaultConfiguredSource = nullptr;
 	QString mDefaultConfiguredSourceName;
 };
 
